burr/core/state.py

```diff
@@ -2,13 +2,52 @@
 import copy
 import dataclasses
 import importlib
+import inspect
 import logging
-from typing import Any, Dict, Iterator, Mapping
+from typing import Any, Callable, Dict, Iterator, Mapping, Union
 
 from burr.core import serde
 
 logger = logging.getLogger(__name__)
 
+FIELD_SERIALIZATION = {}
+
+
+def register_field_serde(field_name: str, serializer: Callable, deserializer: Callable):
```

Comment:
We don't need nested stuff but might we want it at some point? How hard would it be to add?

Reply:
People can reuse the serde module for that if they want. Otherwise not a framework concern.